### PR TITLE
Optimize ignoreFiles patterns

### PR DESCRIPTION
**Pull request by sebakerckhof**
This is a continuation of @c9s work (https://github.com/meteor/meteor/pull/7483) to fix (https://github.com/meteor/meteor/issues/7482)

Merging the different regex to one is quite a bit faster than executing them one-by-one (see: https://github.com/meteor/meteor/pull/7483#issuecomment-249938794 for a rough estimate).

### CODE REVIEW

File: tools/tests/bundle-ignore-files.js [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+  const patterns = bundler.ignoreFiles;
+  const matchingInputs = [
+    '.git/',
+    '.meteor/',
+    '.DS_Store',
+    '.aaabbb.swp',
+    'Thumbs.db'
+  ];
+
+  matchingInputs.forEach(input => selftest.expectEqual(patterns.some(p => p.test(input)),true));
```
Comment (abernix) [inline tools/tests/bundle-ignore-files.js]: Probably a good idea to add some `nonMatchingInputs` and make sure that they fail (`expectEqual`...`false`) so both sides are tested?

Comment (sebakerckhof) [inline tools/tests/bundle-ignore-files.js]: Yeah, i meant to do that at first, but honestly, the regex is so simple it seemed a bit pointless.

Comment (abernix) [inline tools/tests/bundle-ignore-files.js]: Well, funny enough, there's a bug in the regex right now. 🙊  #7787 

:smile:

Comment (sebakerckhof) [inline tools/tests/bundle-ignore-files.js]: Dammit :p.
But still, I mean the concatenated version will match the same things like the separate version and I don't really know how to make a good lists of stuff that doesn't match a regex.

**Comment by sebakerckhof** [comment]
Ok, I added the example from the bug report as one that shouldn't match.

File: tools/tests/bundle-ignore-files.js [hunk 2]
```diff
@@ -12,4 +12,10 @@ selftest.define("bundle-ignore-files", () => {
   ];
 
   matchingInputs.forEach(input => selftest.expectEqual(patterns.some(p => p.test(input)),true));
+
+  const nonMatchingInputs = [
+    '/imports/components/Icon/index.js',
```
Comment (abernix) [inline tools/tests/bundle-ignore-files.js]: This should actually just be `Icon` (this only matches the segment, not the full path).

Can you add `Icon.` to the `matchingInputs` test too?  Truthfully, I have no idea what an `Icon.` file is in the first place and the original commit (63092aabc4067a505ae1c1269e03f7a2b67b68c0) didn't mention, but it seems to be some sort of cache file that should be ignored and we might as well keep with that.

Comment (sebakerckhof) [inline tools/tests/bundle-ignore-files.js]: fixed

Comment (mjmasn) [inline tools/tests/bundle-ignore-files.js]: [This](http://superuser.com/questions/298785/icon-file-on-os-x-desktop) states that the file is actually called `Icon\r`. May need to update that regex again ;)

Comment (abernix) [inline tools/tests/bundle-ignore-files.js]: @mjmasn Wow...what the file!? That was very likely the original intention.  Thanks for finding that.

@sebakerckhof Could you update?

File: tools/isobuild/bundler.js [hunk 3]
```diff
@@ -173,11 +173,13 @@ import { CORDOVA_PLATFORM_VERSIONS } from '../cordova';
 
 // files to ignore when bundling. node has no globs, so use regexps
 exports.ignoreFiles = [
-    /~$/, /^\.#/, /^#.*#$/,  // emacs swap files
-    /^\..*\.sw.$/,  // vim swap files
-    /^\.DS_Store\/?$/, /^ehthumbs\.db$/, /^Icon.$/, /^Thumbs\.db$/,
-    /^\.meteor\/$/, /* avoids scanning N^2 files when bundling all packages */
-    /^\.git\/$/ /* often has too many files to watch */
+    /~$/, /^\.#/,
+    /^(\.meteor\/|\.git\/|Thumbs\.db|\.DS_Store\/?|Icon\.|ehthumbs\.db|\..*\.sw.|#.*#)$/,
```
Comment (abernix) [inline tools/isobuild/bundler.js]: Per https://github.com/meteor/meteor/pull/7835#discussion_r81148872

Could you change `Icon\.` to `Icon\r`?

File: tools/tests/bundle-ignore-files.js [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+const bundler = require('../isobuild/bundler.js');
+
+selftest.define("bundle-ignore-files", () => {
+  const patterns = bundler.ignoreFiles;
+  const matchingInputs = [
+    '.git/',
+    '.meteor/',
+    '.DS_Store',
+    '.aaabbb.swp',
+    'Icon.',
```
Comment (abernix) [inline tools/tests/bundle-ignore-files.js]: Per https://github.com/meteor/meteor/pull/7835#discussion_r81148872

Could you change `Icon.` to `Icon\r`?

File: tools/isobuild/bundler.js (hunk 3 above)
Comment (sebakerckhof) [inline tools/isobuild/bundler.js]: done

**Pull request merged**